### PR TITLE
Implementing Cancel Booking Endpoint

### PR DESCRIPTION
**Pull request by nbdamian**
Add support for cancel booking endpoint as in API Changed Feb 2018

### CODE REVIEW

File: src/APIClient.php [hunk 1]
```diff
@@ -558,4 +558,16 @@ public function api_experiences($supplier_key) {
...
+
+	/**
+	 * @param string $booking_id
+	 *
+	 * @return mixed
+	 * @throws \Rtbs\ApiHelper\Exceptions\ApiClientException
+	 * @throws \Rtbs\ApiHelper\Exceptions\ApiClientNetworkException
+	 */
+	public function api_cancel_booking($booking_id) {
+		return $this->call('/api/booking/'.$booking_id.'/cancel');
```
Comment (bumperbox) [inline src/APIClient.php]: need to urlencode the booking id, as it can contain# symbol

Comment (nbdamian) [reply src/APIClient.php]: I have added urlencode in 07120aaab9a9d8cf7b568976f599a4cf102b5deb

**Comment by markcww** [comment]
Hi @nbdamian, thanks for your contributions. I'd suggest renaming booking_id to token_or_booking_id as both are accepted, and this method is comparable to methods like booking status. 

Please note as currently implemented, an attempt to cancel a booking is prevented if the booking was paid via the RTBS payment gateway.

**Comment by nbdamian** [comment]
> Hi @nbdamian, thanks for your contributions. I'd suggest renaming booking_id to token_or_booking_id as both are accepted, and this method is comparable to methods like booking status.
> 
> Please note as currently implemented, an attempt to cancel a booking is prevented if the booking was paid via the RTBS payment gateway.

I had attempted to use the booking_token, however was receiving an empty response from the API. Hence why I am using booking_id.

Regarding the cancel a booking being prevented, will this return an error, or will it simply not change the booking status?

**Comment by markcww** [comment]
It'll return an error "err-cancel-unsupported".

**Pull request merged**